### ISSUE TITLE
Search box at top-right corner (or Ctrl + K) causes screen to freeze.

### ISSUE THREAD

**Issue by dmitrilc** [vaadin/docs#1451]
Performing queries using the search box causes jank. 

I have attached both Chrome and Firefox performance profiles.

[Firefox 2022-06-03 08.19 profile.json.gz](https://github.com/vaadin/docs/files/8855519/Firefox.2022-06-03.08.19.profile.json.gz)
[Chrome profile and video.zip](https://github.com/vaadin/docs/files/8855525/Chrome.profile.and.video.zip)

Example of other websites that don't have this issue:
https://kotlinlang.org/docs/home.html
https://www.typescriptlang.org/docs/handbook/
https://docs.oracle.com/en/java/javase/17/docs/api/index.html

**Comment by tarekoraby** [comment]
I can reproduce this. There is around a one-second freeze if I keep entering characters in the search field.

**Comment by jouni** [comment]
The longer the search string, the longer it takes for Fuse.js to return the results. Not sure if it’s because of the size of our search index, or the configuration we pass to Fuse.

**Comment by mvysny** [comment]
Yup, after you accidentally paste something like this to the search box, the page is pretty much dead: `https://vaadin.com/docs/latest/routing/lifecycle#beforeenterevent`.
